src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py

```diff
@@ -1338,19 +1439,189 @@ def main():
+        # actually contains a field qu0, which is never used for restart (the qu0 can be obtained
+        # as the qd0 from the topologically upstream segments, just like during the calculation).
+        # In any case, the qu0 currently in the WRF-Hydro output is populated with the same value
+        # as the qd0.
+        #[pd.DataFrame(d[:,-3::2], index=i, columns=["qd0", "h0"]) for i, d in run_results],
+        #[pd.DataFrame(r[1][:,-3:], index=r[0], columns=["qu0", "v0", "h0"]) for r in run_results],
+        [pd.DataFrame(r[1][:,[-3,-3,-1]], index=r[0], columns=["qu0", "qd0", "h0"]) for r in run_results],
+        copy=False,
+    )
+
+
+if __name__ == "__main__":
+
+    (
+        log_parameters,
+        supernetwork_parameters,
+        waterbody_parameters,
+        compute_parameters,
+        forcing_parameters,
+        restart_parameters,
+        output_parameters,
+        parity_parameters,
+        data_assimilation_parameters,
+        diffusive_parameters,
+    ) = _input_handler()
+
+    verbose = log_parameters.get("verbose", None)
+    showtiming = log_parameters.get("showtiming", None)
+    debuglevel = log_parameters.get("debuglevel", 0)
+
+    if showtiming:
+        main_start_time = time.time()
+
+    (
+        connections,
+        param_df,
+        wbodies,
+        waterbodies_df,
+        break_network_at_waterbodies,
+        independent_networks,
+        reaches_bytw,
+        rconn,
+    ) = nwm_network_preprocess(
+        supernetwork_parameters,
+        waterbody_parameters,
+        showtiming=showtiming,
+        verbose=verbose,
+        debuglevel=debuglevel,
+    )
+
+    waterbodies_df, q0 = nwm_initial_warmstate_preprocess(
+        break_network_at_waterbodies,
+        restart_parameters,
+        param_df,
+        waterbodies_df,
+        segment_list=None,
+        wbodies_list=None,
+        showtiming=showtiming,
+        verbose=verbose,
+        debuglevel=debuglevel,
+    )
+
+
+    # The inputs below assume a very pedantic setup
+    # with each run set explicitly defined, so...
+    # TODO: Make this more flexible.
+    run_sets = forcing_parameters.get("qlat_forcing_sets", False)
+
+    # TODO: Data Assimilation will be something like the parity block
+    # if DA:
+    #     da_sets = [BIG LIST OF DA BLOCKS]
+
+    if "wrf_hydro_parity_check" in output_parameters:
+        parity_sets = parity_parameters.get("parity_check_compare_file_sets", False)
+    else:
+        parity_sets = []
+
+    parallel_compute_method = (compute_parameters.get("parallel_compute_method", None),)
+    subnetwork_target_size = (compute_parameters.get("subnetwork_target_size", 1),)
+    cpu_pool = (compute_parameters.get("cpu_pool", None),)
+    qts_subdivisions = forcing_parameters.get("qts_subdivisions", 1)
+
+    qlats, usgs_df = nwm_forcing_preprocess(
+        connections,
+        break_network_at_waterbodies,
+        run_sets[0],
```

Comment:
This is confusing -- we have both functional returns as well as a modified input value (run_sets[0] is populated with additional dictionary elements)